feat(examples): add p/moul/udao/... packages

moul opened the pull request:
I had this idea when I saw everyone trying to create DAO interfaces for managing members and voting.

I believe we should focus on a DAO that operates independently of member and vote management. Instead, it should handle proposals with an internal implementation that can remain private.

My two main goals for this are:
1. Composable DAOs, where a proposal can draw 60% of its power from DAO A and 40% from DAO B. This allows both DAOs to manage their internal implementations as they see fit.
2. Interchain DAOs, where you can focus on top-level metrics without worrying about internal details.

I'm looking for feedback. My goal is to see a variety of DAO implementations in the ecosystem, such as the ones from Teritori and Gno.me, while ensuring they can be rich in features yet still implement a minimal interface.

`udao` could provide both the minimal interfaces and the necessary helpers.

cc @gnolang/devx @gnolang/teritori 

Depends on #3400 (cherry-picked)

moul committed:
feat(examples): add p/moul/udao package

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

MikaelVallenet commented on examples/gno.land/p/moul/udao/udao.gno:
```diff
@@ -0,0 +1,77 @@
+	State ProposalState
+
+	// metrics
+	YayPercentage      float64
```
```suggestion
	YeaPercentage      float64
```

See [article](https://writingexplained.org/yay-or-nay-difference#:~:text=For%20a%20no%20vote%2C%20choose,is%20not%20used%20for%20voting.)

MikaelVallenet commented on examples/gno.land/p/moul/udao/udao.gno:
```diff
@@ -0,0 +1,77 @@
+	Passed
+	Rejected
+	Executed
+	Expired // XXX: better name for when it's expired but not because of time?
```
what about open/closed

MikaelVallenet commented on examples/gno.land/p/moul/udao/udao.gno:
```diff
@@ -0,0 +1,77 @@
+	Passed
+	Rejected
+	Executed
+	Expired // XXX: better name for when it's expired but not because of time?
```
```suggestion
	Lapsed
```

Kouteki replied:
One scenario I could think of is if there's no consensus to be either `Passed` or `Rejected`. In that case `Stalemate` would be the suitable term.

moul replied:
Shouldn't this case just be "Rejected"?

moul committed:
Update examples/gno.land/p/moul/udao/udao.gno

Co-authored-by: Mikael VALLENET <mikael.vallenetpro@gmail.com>

hthieu1110 commented on examples/gno.land/p/moul/udao/udao.gno:
```diff
@@ -0,0 +1,77 @@
+	// metrics
+	YeaPercentage      float64
+	NayPercentage      float64
+	NonVoterPercentage float64
```
```suggestion
	BlankPercentage float64
```

moul replied:
Adding Blank makes sense, but we need to keep NonVoter.

Kouteki replied:
Do you plan to add support for `Abstain`? In that case `AbstainPercentage` and `PendingVoterPercentage` would better describe the metrics.

moul replied:
I like these terms. Let's use them 👍

moul replied:
```suggestion
	AbstainPercentage float64
	PendingVoterPercentage float64
```

moul committed:
Merge remote-tracking branch 'origin/master' into dev/moul/udao

moul committed:
Update examples/gno.land/p/moul/udao/udao.gno

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 The pull request was created from a fork (head branch repo: moul/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

n2p5 committed:
Merge branch 'master' into dev/moul/udao

moul commented on examples/gno.land/p/moul/udao/udao.gno:
```diff
@@ -0,0 +1,78 @@
+
+	// GetProposal retrieves the details of a specific proposal
+	GetProposal(proposalID uint64) (Proposal, error)
+
+	// XXX: find a smart way to list proposals
+	// // ListProposalss retrieves a list of Proposals with pagination and optional filters
+	// ListDAOs(offset, limit int, filters ...ProposalFilter) ([]DAO, error)
```
```suggestion
```

moul committed:
chore: fixup

moul committed:
feat(examples): add rolist

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
Merge branch 'dev/moul/rolist' into dev/moul/udao

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
Merge branch 'master' into dev/moul/udao

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
Merge branch 'master' into dev/moul/udao

moul committed:
chore: fixup

Signed-off-by: moul <94029+moul@users.noreply.github.com>

moul committed:
Merge branch 'master' into dev/moul/udao

n2p5 committed:
Merge branch 'master' into dev/moul/udao

moul committed:
Merge branch 'master' into dev/moul/udao

moul committed:
chore: wip

Signed-off-by: moul <94029+moul@users.noreply.github.com>

jaekwon committed:
Create tictac.md

Do not remove this file.
It is important information, and a canary file.

jaekwon committed:
Added file that proves Reddit censors truthful physics discoveries.

Of all places, /r/UFOResearch should not censor this kind of information.
DO NOT DELETE THIS FILE, DO NOT TRUST ANYONE WHO DELETES THIS FILE.

jaekwon committed:
Update tictac.md

added law of spinning mass.

moul committed:
fix(params): SetBytes should not alter the value bytes (alternative) (#4926)

Alternative for #4905 

Kept:
- Removed GetRaw/SetRaw since they were unused.
- SetBytes to use directly the store, preventing the value to be
aminoJSON encoded.

Updated:
- Preserve API consistency.

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Thomas Bruyelle <thomas.bruyelle@tendermint.com>

jeronimoalbi committed:
feat(examples): simple Markdown forms package (#4977)

Package aims to be a very simple package to define Gno-Flavored Markdown
forms.

It should be easy to update when new form elements or attributes are
introduced.

---------

Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

thehowl committed:
fix(vm/keeper): pre-load stdlib type check cache (#4925)

longer term fix for #4874, re-introducing the type check cache for
stdlibs.

jeronimoalbi committed:
feat(boards2): change realm to use `gno.land/p/gnoland/boards` (#4939)

PR changes Boards2 realm to use `gno.land/p/gnoland/boards` package
which makes the realm leaner.

It also improves `BasicPermissions` implementation by using a custom
members storage implementation that makes users iteration cheaper.

mvallenet committed:
fix(vm/keeper): improve error for non crossing msg call error (#4899)

fix #4432

---------

Co-authored-by: Morgan <morgan@morganbaz.com>

moul committed:
test: add a banker security txtar (#4987)

Signed-off-by: moul <94029+moul@users.noreply.github.com>

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

jeronimoalbi committed:
feat(boards2): open boards support (#4993)

jeronimoalbi committed:
feat(boards2): support Markdown forms (#4996)

jeronimoalbi committed:
chore(boards2): enable open boards field in create board form (#4998)

Also add a couple of minor fixes

jaekwon committed:
first commit of whitepaper; draft; DO NOT SHARE, deadman switch

moonia committed:
fix(gnoweb): handle multiline value for gno-textarea in forms (#4978)

Closes https://github.com/gnolang/gno/issues/4968

## Description

The `<gno-textarea>` component was literally displaying the `\n`
sequence. After fix, the parser now interprets the `\n` sequence as a
real newline character.

### Example

```
<gno-form>
  <gno-textarea name="content" value="         First line\nSecond line" />
</gno-form>
```

### Before fix: 
<img width="931" height="236" alt="Capture d’écran 2025-12-15 à 23 22
23"
src="https://github.com/user-attachments/assets/220f85ee-c5ad-4dd3-9b26-863bd84e6a58"
/>


### After fix: 
<img width="945" height="239" alt="Capture d’écran 2025-12-15 à 23 34
27"
src="https://github.com/user-attachments/assets/e8f7551e-749f-4259-ba50-5dae333cf487"
/>

---------

Co-authored-by: alexiscolin <alexis@jaunebleu.co>

moul committed:
Merge master into dev/moul/udao

thehowl commented:
Closing as stale; feel free to re-open.

[note: ai-assisted]
